tests/Unit/Evolution/Systems/NewtonianEuler/Test_Characteristics.cpp

```diff
@@ -292,6 +268,61 @@ void test_right_and_left_eigenvectors() noexcept {
   }
 }
 
+template <size_t Dim>
+void test_right_and_left_eigenvectors() noexcept {
+  // This test verifies that the eigenvectors satisfy the conditions by which
+  // they are defined:
+  // - the right and left eigenvectors are matrix inverses of each other, i.e.,
+  //   left * right == right * left == identity
+  // - the right and left eigenvectors diagonalize the flux Jacobian, i.e.,
+  //   right * eigenvalues * left == flux
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> distribution(-1., 1.);
+  std::uniform_real_distribution<> distribution_positive(1e-3, 1.);
+
+  const auto nn_generator = make_not_null(&generator);
+  const auto nn_distribution = make_not_null(&distribution);
+  const auto nn_distribution_positive = make_not_null(&distribution_positive);
+
+  const double used_for_size = 0.;
+  // This computes a unit normal. It is NOT uniformly distributed in angle,
+  // but for this test the angular distribution is not important.
+  const auto unit_normal = [&]() noexcept {
+    auto result = make_with_random_values<tnsr::i<double, Dim>>(
+        nn_generator, nn_distribution, used_for_size);
+    double normal_magnitude = get(magnitude(result));
+    // Though highly unlikely, the normal could have length nearly 0. If this
+    // happens, we edit the normal to make it non-zero.
+    if (normal_magnitude < 1e-3) {
+      get<0>(result) += 0.9;
+      normal_magnitude = get(magnitude(result));
+    }
+    for (auto& n_i : result) {
+      n_i /= normal_magnitude;
+    }
+    return result;
+  }
+  ();
+
+  // To check the diagonalization of the Jacobian, we need a self consistent set
+  // of primitive and derived-from-primitive variables -- so generate everything
+  // from the primitives
+  const auto density = make_with_random_values<Scalar<double>>(
+      nn_generator, nn_distribution_positive, used_for_size);
+  const auto velocity = make_with_random_values<tnsr::I<double, Dim>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto specific_internal_energy = make_with_random_values<Scalar<double>>(
+      nn_generator, nn_distribution_positive, used_for_size);
+
+  const EquationsOfState::IdealFluid<false> ideal_gas_eos{5. / 3.};
+  test_left_and_right_eigenvectors_impl(
+      density, velocity, specific_internal_energy, unit_normal, ideal_gas_eos);
+
+  const EquationsOfState::PolytropicFluid<false> polytropic_eos{100., 2.};
+  test_left_and_right_eigenvectors_impl(
+      density, velocity, specific_internal_energy, unit_normal, polytropic_eos);
+}
+
```

Comment:
It might be worth checking along coordinate axes as well, for which you'd like to loop over directions, as done for the speeds.

Reply:
Good idea, I'll add a new commit to this PR that checks this